javalib-ex/src/main/scala/java/io/DataInputStream.scala

```diff
@@ -106,10 +94,12 @@ class DataInputStream(in: InputStream) extends FilterInputStream(in)
   }
 
   def readInt(): Int = {
-    if (hasArrayBuffer)
+    if (hasArrayBuffer) {
       bufDataView.getInt32(consumePos(4))
-    else
-      view(4).getInt32(0)
+    } else {
+      ((readByte() & 0xff) << 24) | ((readByte() & 0xff) << 16) |
+      ((readByte() & 0xff) << 8) | (readByte() & 0xff)
```

Comment:
Consider calling `readUnsignedByte()` instead of `(readByte() & 0xff)` to make the intent more explicit. This comment applies to other `readXYZ()` methods.